Add blog post about perforce EULA (#471)
voxpupuli/voxpupuli.github.io: _posts/2025-05-16-perforce-eula.md

---
layout: post
title: The Perforce EULA and impact on Vox Pupuli
date: 2025-05-16
github_username: bastelfreak
---

> In November 2024, [Perforce announced](https://www.puppet.com/blog/open-source-puppet-updates-2025) that they will stop providing most tools & projects as open source.

Reviewer: We might consider mentioning that while the repos remain available and unarchived, they've made it clear they are not going to maintain them at all and we've seen zero evidence of backporting any fixes from Puppet Core whatsoever.

Author: I'm a bit careful here. They also said that the apt/yum repos are read only but then they still published new puppet-release packages with an updated gpg key. I've zero confidence anymore in statements from perforce.

> In early 2025, Puppet will begin to ship any new binaries and packages developed by our team to a private, hardened, and controlled location.
> Community contributors will have free access to this private repo under the terms of an End-User License Agreement (EULA) for development use. There will be no license changes for the open source version of Puppet.

> As the top community contributors to the Puppet module ecosystem and the maintainers of much of the Puppet module testing pipeline tooling, Vox Pupuli has significant expertise in developer experience.

Reviewer: ```suggestion
As the top community contributors to the Puppet module ecosystem and the maintainers of much of the Puppet module testing pipeline tooling, Vox Pupuli has significant expertise in the developer experience.
```

> The Vox Pupuli PMC immediately reached out to Perforce to discuss the [provided](https://github.com/voxpupuli/community-triage/issues/29) "[PUPPET® CORE DEVELOPER PROGRAM LICENSE AGREEMENT](https://www.perforce.com/system/files/2025-02/Puppet-Core-Developer-Program-License-Agreement.pdf)" in hopes of improving it and reducing the friction to contributing Puppet modules.

Reviewer 2: ```suggestion
The [Vox Pupuli PMC](https://voxpupuli.org/elections/) immediately reached out to Perforce to discuss the [provided](https://github.com/voxpupuli/community-triage/issues/29) "[PUPPET® CORE DEVELOPER PROGRAM LICENSE AGREEMENT](https://www.perforce.com/system/files/2025-02/Puppet-Core-Developer-Program-License-Agreement.pdf)" in hopes of improving it and reducing the friction to contributing Puppet modules.
```

> As of today (2025-05-16), Vox Pupuli is unable to sign the current version of the Puppet Core Developer EULA.

Reviewer: Maybe `the Vox Pupuli community has reached the conculsion that we are unable to sign`.  It sounds a bit like they did something today that caused us to not sign it.

> The restrictions placed on usage prevent effective testing and distribution of modules and we don’t want to expose ourselves to potential legal challenges.

Reviewer: We may want to provide more details about what exactly we have a problem with.  I'd read this as an end user and be left wondering what the problems actually are and why I should agree that this is the only viable path.

Reviewer 3: I’d rather people read the thread and saw what we tried to get addressed

Unfortunately, this means that Vox Pupuli modules are not tested against Puppet Core, so please use them at your own risk.

> We are still interested in working together with Perforce, but their current EULA doesn't allow it.

Reviewer: ```suggestion
We are still interested in working together with Perforce, and would prefer that modules were tested against both OpenVox and Puppet Core, but their current EULA makes this untenable.
```

> Vox Pupuli will continue to test all of our modules against the [Open Vox packeges](https://voxpupuli.org/openvox/).

Reviewer 4: "packages"

Reviewer 5: That's a typo I had in my typo suggestion fix in https://github.com/voxpupuli/voxpupuli.github.io/pull/471#discussion_r2093592144 before I fixed it :(

Reviewer 4: I guess the fix introduced a new typo?

Reviewer 5: > (I don't even know how to suggest changes the way you did there see🙈)

GitHub has the `Add a suggestion` button with ctrl+g as the shortcut. Doesn't always work that well, but I regularly type it manually. Essentially it's a regular code block with `suggestion:
````
```suggestion
````

date: 2025-05-19
github_username: bastelfreak
---

In November 2024, [Perforce announced](https://www.puppet.com/blog/open-source-puppet-updates-2025) that they will stop providing most tools & projects as open source.

> In early 2025, Puppet will begin to ship any new binaries and packages developed by our team to a private, hardened, and controlled location.
> Community contributors will have free access to this private repo under the terms of an End-User License Agreement (EULA) for development use. There will be no license changes for the open source version of Puppet.

As the top community contributors to the Puppet module ecosystem and the maintainers of much of the Puppet module testing pipeline tooling, Vox Pupuli has significant expertise in the developer experience.
The [Vox Pupuli PMC](https://voxpupuli.org/elections/) immediately reached out to Perforce to discuss the [provided](https://github.com/voxpupuli/community-triage/issues/29) "[PUPPET® CORE DEVELOPER PROGRAM LICENSE AGREEMENT](https://www.perforce.com/system/files/2025-02/Puppet-Core-Developer-Program-License-Agreement.pdf)" in hopes of improving it and reducing the friction to contributing Puppet modules.

> The Vox Pupuli community has reached the conculsion that we are unable to sign the Puppet Core Developer EULA.

Reviewer 5: ```suggestion
The Vox Pupuli community has reached the conclusion that we are unable to sign the Puppet Core Developer EULA.
```
